crates/ruff_workspace/src/configuration.rs

```diff
@@ -1088,31 +1080,27 @@ impl LintConfiguration {
 
     #[must_use]
     pub fn combine(self, config: Self) -> Self {
+        let mut rule_selections = config.rule_selections;
+        rule_selections.extend(self.rule_selections);
+
+        let mut extend_safe_fixes = config.extend_safe_fixes;
+        extend_safe_fixes.extend(self.extend_safe_fixes);
+
+        let mut extend_unsafe_fixes = config.extend_unsafe_fixes;
+        extend_unsafe_fixes.extend(self.extend_unsafe_fixes);
+
+        let mut extend_per_file_ignores = config.extend_per_file_ignores;
+        extend_per_file_ignores.extend(self.extend_per_file_ignores);
```

Comment:
Another unrelated config change. It avoids allocating a new `Vec` and instead reuses the existing vector.